lib/src/watch_it_state.dart

```diff
@@ -251,11 +251,10 @@ class _WatchItState {
         /// Only if this isn't used to register a handler
         ///  still the same stream so we can directly return lastvalue
         if (handler == null) {
-          assert(watch.lastValue != null && watch.lastValue!.data != null);
+          assert(watch.lastValue != null && !watch.lastValue!.hasError);
```

Comment:
In the case when the Stream has a valid null value, `hasData` will be false and `hasError` will also be false. This check will ensure there is no error, and the next line will cast the value as R (which will accept `null` if R itself is nullable. But will not accept `null` if R is not nullable).

I'm not sure if you want a more vigorous check to ensure there is data when R is not nullable. such as 

```dart
assert(!(null is R) || (watch.lastValue!.hasData));
```